### DISCUSSION TITLE
How to copy HTML source and paste as markdown text?

### DISCUSSION THREAD

**Discussion by ooker777** [hluk/CopyQ#2005] [Q&A]
In my understanding, the API of CopyQ use [Qt Script](https://doc.qt.io/qt-5/qtscript-index.html), which supports ECMAScript, which is basically JS? There is also [an HTML to Markdown converter written in JavaScript](https://github.com/mixmark-io/turndown). Can we combine the two?

**Comment by dbareis** [comment]
You are correct that it is "basically" JS but it doesn't support everything the latest javascript does. I was told copyq is ECMAScript 4/5 (whatever that means).

I assume there would be some work getting it going.  I write all my JS commands in files outside of copyq and "include" them with the source command, as it is very painful to use the inbuilt editor.  To get to commands press "F6" while copyq open.
